include/rocksdb/db.h

```diff
@@ -976,6 +977,20 @@ class DB {
     return IngestExternalFile(DefaultColumnFamily(), external_files, options);
   }
 
+  // StartTrace/EndTrace the major workload of the DB
+  virtual Status  StartTrace(
+      std::unique_ptr<TraceWriter>&& trace_writer) = 0;
+  virtual Status StartTrace(const std::string& trace_filename) = 0;
+  virtual Status EndTrace() = 0;
+
+  // StartReplay will replay the trace to the DB opened by trace_reader
+  virtual Status StartReplay(std::vector<ColumnFamilyHandle*>& handles,
+                             std::unique_ptr<TraceReader>&& trace_reader,
+                             bool no_wait = false) = 0;
+  virtual Status StartReplay(std::vector<ColumnFamilyHandle*>& handles,
+                             const std::string& trace_filename,
+                             bool no_wait = false) = 0;
```

Comment:
It is better for StartReplay() to be a standalone function, rather than method of DB. It should be able to call appropriate DB interface. With this interface, any DB (say blobDB, TTLDB) has to implement this function, which is very impractical. I also don't see why this is virtual so that users want to overwrite it.